cmp/filter/FilterChooserModel.ts

```diff
@@ -423,25 +423,25 @@ export class FilterChooserModel extends HoistModel {
 
     @action
     setFavorites(favorites: Filter[]) {
-        this.favorites = favorites.filter(f => this.validateFilter(f));
+        this.favorites = favorites.filter(f => this.validateFilter(f)) as FilterChooserFilter[];
```

Comment:
Had been hoping that `validateFilter` type guard signature would flow through the array `filter` here, but does not appear to be the case, hence the explicit `as`.  LMK if I'm doing it wrong or we can think of a more elegant solution 😄

Reply:
Looks like the issue here is that you are passing a closure (which calls the typeguard) rather than passing the typeguard directly to `filter`.  The closure isn't typed specifically, so TS seems to infer that its return type is `boolean`.

Possible solution would be to pass the typeguard directly, like this
`this.favorites = favorites.filter(this.validateFilter)`.

Would need to ensure `validateFilter` is still bound to the instance on which it's called, since it calls methods on `this`.
  
<img width="397" alt="Screenshot 2024-02-15 at 2 37 43 PM" src="https://github.com/xh/hoist-react/assets/88904581/b2f5c189-769d-42e9-9d7c-e9db22a87102">

Reply:
Thank you for the explanation!  

Not in love with it, but for this little filter seemed worth it to swap to:

`this.favorites = favorites.filter(this.validateFilter.bind(this));`

to leverage the type guard directly (almost) as I had intended.